core/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java

```diff
@@ -305,17 +314,37 @@ public static FileInfo fromXContent(XContentParser parser) throws IOException {
                     }
                 }
             }
+
             // Verify that file information is complete
             if (name == null || Strings.validFileName(name) == false) {
                 throw new ElasticsearchParseException("missing or invalid file name [" + name + "]");
             } else if (physicalName == null || Strings.validFileName(physicalName) == false) {
                 throw new ElasticsearchParseException("missing or invalid physical file name [" + physicalName + "]");
             } else if (length < 0) {
                 throw new ElasticsearchParseException("missing or invalid file length");
+            } else if (checksum == null) {
+                if (physicalName.startsWith("segments_")) {
```

Comment:
can we verify `writtenBy` here too ie. if writtenBy is >= 6.0 we must fail since we are guaranteed to have the checksum here?